### ISSUE TITLE
Restrict tests to our add-ons

### ISSUE THREAD

**Issue by ajtucker** [GSS-Cogs/dd-cms#92]
`yarn test` appears to run the govuk-react-jsx tests, which appear to fail on some documentation requirement. We should be able to just run the tests on our own addons, e.g. `yarn test src/addons/volto-govuk-theme` etc.

Also, our tests currently fail due to some issue with the jest-junit reporter?

**Comment by ajtucker** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/cogs-dd-cms-61e83b63053eb70011d37b4f/issues/GSS-Cogs/dd-cms/92?planning-poker) with ZenHub @charlesons @cjfairchild @osamede20

**Comment by cjfairchild** [comment]
Pull request submitted with same/similar name

**Comment by nosnickid** [comment]
Merging main into this was just a big conflict in the Jenkinsfile from main.

Alex modified the Jenkinsfile to use a docker "agent", and callum modified it to use shell commands to interact with docker directly.

The main change here was to copy out the `junit.xml` file. place it in its own directory, and run the `publishHTML` only on that dir. 

I've tweaked it slightly to use dir() in the Jenkinsfile. Happy to see this merged now.

In running this on my own jenkins server, I've realised the tweaks we made to the `Dockerfile-tests` that try to use docker layer caching aren't helping; jenkins immediately bind-mounts the workspace over any files installed during  the container build.  I'll put some more details into its own issue.
